base/compiler/tfuncs.jl

```diff
@@ -1169,6 +1169,29 @@ function builtin_tfunction(@nospecialize(f), argtypes::Array{Any,1},
     return tf[3](argtypes...)
 end
 
+# Query whether the given intrinsic is nothrow
+intrinsic_nothrow(f::IntrinsicFunction) = !(
+        f === Intrinsics.checked_sdiv_int ||
+        f === Intrinsics.checked_udiv_int ||
+        f === Intrinsics.checked_srem_int ||
+        f === Intrinsics.checked_urem_int
+    )
```

Comment:
Should this include `cglobal`? I know with the rest of the code it's guarded by `is_pure_intrinsic_infer`, but maybe we should include it to make the function as accurate as possible.

Reply:
Yep, fair point.